### ISSUE TITLE
Spike - Look into Infusion

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-accessibility-passport#123]

**Comment by gobyrne** [comment]
https://fluidproject.org/infusion.html

**Comment by mnigh** [comment]
- text size
- text style
- line spacing
- contrast (future)
- table of contents

**Comment by mnigh** [comment]
The Infusion web framework is built on top of jQuery which is quite heavy and would be in addition to React, which is not a great pattern, and would have to be maintained separately from the React components that we are building.

The library calls the UI via jQuery in an HTML file and calls in as an [iframe](https://github.com/fluid-project/build.fluidproject.org/blob/main/index.html#L59), which seems to get [crawled](https://www.google.com/search?q=SeparatedPanelPrefsEditorFrame.html&oq=SeparatedPanelPrefsEditorFrame.html&aqs=chrome.0.69i59l2j69i60l3.124j0j7&sourceid=chrome&ie=UTF-8) by Google.

@JoshBeveridge Unless you see more value in this tool than I, what would be the alternative, creating our own components that you would design?

**Comment by JoshBeveridge** [comment]
Yeah, if this is built on jQuery, I'd lean towards saying it's a no go. 

The only alternative I can see is building our own, and the writing JavaScript to customize CSS variables through Hydrogen. As mentioned in our meeting earlier today, `font-size`, `font-family`, `line-spacing`, and the table of contents are low hanging fruit with this approach. Skinning a website with multiple different contrast themes is significantly more work, however, and it would require us to design each of our components multiple times to ensure that they had matching contrast versions. Think dark mode, but times it by 5-6. 

I see there being unintended side effects with us modifying `line-height` in Hydrogen due to it building the entire site's vertical rhythm off of the line-height (this would mean that changing the line-height would also change values for margin, padding, etc.).

Worth noting too that following this route would mean changing our tech stack, and likely rebuilding what we have so far from scratch in Infusion. I don't believe migrating to older tech just for the contrast themes is enough of a reason. As @RM-1978 mentioned, because we'll have dark mode at our disposal, pairing it with the other options described above might be enough for now.

**Comment by mnigh** [comment]
@gobyrne is there action that you wanted to happen out of this spike or can it be closed since Infusion is not a good option and Hydrogen is not yet ready to be used for creating components for this type of manipulation?
